benchmarks/benchmarks/selections.py

```diff
@@ -34,3 +34,51 @@ def time_simple_selections(self, selection_string):
             self.u.select_atoms(selection_string)
         else:
             self.u.selectAtoms(selection_string)
+
+class GeoSelectionBench(object):
+    """Benchmarks for the various MDAnalysis
+    geometric selection strings.
+    """
+
+    # all selection strings verified
+    # to produce non-zero atom groups
+    # with GRO test file
+    params = (['around 5.0 resid 1',
+               'sphlayer 2.4 6.0 (protein)',
+               'sphzone 6.0 (protein)',
+               'cylayer 5 10 10 -8 protein',
+               'cyzone 15 4 -8 protein',
+               'point 5.0 5.0 5.0 3.5',
+               'prop z >= 5.0',
+               'prop abs z <= 5.0'],
+              [True, False], # updating flags
+              [[False, True], [True, False]]) # periodic flags
+
+
+    # benchmarks should include static &
+    # dynamic selections & periodic
+    # vs non-periodic
+    param_names = ['selection_string',
+                   'dynamic_selection',
+                   'periodic_selection']
+
+
+    def setup(self,
+              selection_string,
+              dynamic_selection,
+              periodic_selection):
+        self.u = MDAnalysis.Universe(GRO)
+
+    def time_geometric_selections(self,
+                                  selection_string,
+                                  dynamic_selection,
+                                  periodic_selection):
+
+        # set core flags for PBC accounting
+        MDAnalysis.core.flags['use_periodic_selections'] = periodic_selection[0]
+        MDAnalysis.core.flags['use_KDTree_routines'] = periodic_selection[1]
```

Comment:
Maybe not so good. We were thinking of removing those flags that act globally on the library.

Reply:
What's the preferred equivalent then?

Reply:
There *should* be keywords available to switch on PBC. If there are not, raise issue... and reference #782 .

Reply:
Actually, there is #759 open: add kwargs for PBC/no-PBC.